pkg/power/components/source/rapl_sysfs_util.go

```diff
@@ -19,44 +19,39 @@ package source
 import (
 	"fmt"
 	"os"
+	"os/exec"
 	"strconv"
 	"strings"
-
-	"k8s.io/klog/v2"
 )
 
-func getNumCPUs() int {
-	data, err := os.ReadFile(cpuInfoPath)
+func getNumPackage() (int, error) {
+	grep := exec.Command("grep", "Socket")
```

Comment:
Can we avoid the use of bash commands? It's better to use libraries instead of bash commands since library results in more transparency, efficiency, and better code design.

Furthermore, bash commands are not the most efficient approach. In this particular case, you are running two commands: `lscpu` and `grep`. The `grep` command is traversing a file as the approach that I suggested before, so we can simply open the file and read it directly.

Reply:
Also, we should save the num package in a global variable so that we don't need to traverse a file every time that we call this function.

Reply:
OK,  @marceloamaral, I used to understand your point that the code style like using "bash -c" in Golang is bad because it has an assumption that the system has installed `bash`, but `lscpu` and `grep` should be included in any Linux distro.

Now I think I totally got your points! The bad news is that not only the `getS390xArchitecture()` directly called `exec.Command`, but also for `getArm64Architecture()` and `getX86Architecture()`... Maybe more other places...
It seems to be a tradeoff between code independency and code simplicity.

Another question is where should be the best place for us to save such project-wide global variables and APIs? As I mentioned above, `config` pkg might not be very appropriate to act as such role, since they are configuration related...

Maybe this topic is a little bit leaving the current PR's title:) Shall we close current one and open a new issue to trace similar refactoring efforts throughout the whole project code?

Reply:
@jiere, let's fix the RAPL issues with this PR first and then we refactor the other codes later. But we should replace the existing `getNumPackage` function with a code that reads the `/proc/cpuinfo` file. So we move a step torwards this direction.

In a separate PR, we can create a new package called `runtime` at the same level as `config`. The  `runtime` package will be responsible for providing information about the runtime system, including the CPU topology, model, cgroup version, etc. We can also fix the other codes this new PR, if possible.

Reply:
Regarding the Library to read the runtime system information, we can use https://github.com/jaypipes/ghw#cpu